src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -185,6 +199,31 @@ export class LogStreamRegistry {
     public hasTextEditor(uri: vscode.Uri): boolean {
         return this.hasLog(uri) && this.getTextEditor(uri) !== undefined
     }
+
+    public getStreamIdMap(uri: vscode.Uri): Map<number, string> | undefined {
+        return this.activeLogs.get(uriToKey(uri))?.streamIds
+    }
+
+    private setRangeForStreamIdMap(uri: vscode.Uri, lowerBound: number, upperBound: number, streamID: string): void {
+        // Note: Inclusive of both bounds.
+        let currentLine = lowerBound
+        while (currentLine <= upperBound) {
+            this.setStreamIdMap(uri, currentLine, streamID)
+            currentLine += 1
+        }
```

Comment:
Minor change for readability:
```suggestion
        for (let currentLine = lowerBound; currentLine <= upperBound; currentLine++) {
            this.setStreamIdMap(uri, currentLine, streamID)
        }
```